test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -521,6 +527,10 @@ public final void testDeprecatedBoost() throws IOException {
         assertParseMinimalWarnings();
     }
 
+    protected IndexVersion boostNotAllowedIndexVersion() {
```

Comment:
There is the problem that checking for boosts not allowed caused sparse_vector mappings to error because sparse_vector is not allowed for 8.0.0.

I've created an overrridable method that returns the index version to test boosting for.

I'm happy to hear about other ideas for solving this in a simpler way 👍